### PR TITLE
feat(core): add last use time to user mfa verifications

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add `lastUsedAt` to `user.mfaVerifications`, when a auth factor is verified, it will be updated with current time.

And sort `availableFactors` when throw error.

## Testing

Unit tests.

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/interaction/mfa.ts [hunk 1]
```diff
@@ -121,6 +121,21 @@ export default function mfaRoutes<T extends IRouterParamContext>(
...
+          if (mfa.id !== verifiedMfa.id) {
+            return mfa;
+          }
+
+          return {
+            ...mfa,
+            lastUsedAt: new Date().toISOString(),
+          };
+        }),
+      });
```
Comment (simeng-li) [inline packages/core/src/routes/interaction/mfa.ts:137]: Could use a single query method? `updateUserMfaVerificationsById(userId, mfaId)`. Looks like no need to query user then update.

File: packages/core/src/routes/interaction/verifications/mfa-payload-verification.ts [hunk 2]
```diff
@@ -232,7 +232,7 @@ export async function verifyMfaPayloadVerification(
       // Update the authenticator's counter in the DB to the newest count in the authentication
       await tenant.queries.users.updateUserById(accountId, {
         mfaVerifications: user.mfaVerifications.map((mfa) => {
-          if (mfa.type !== MfaFactor.WebAuthn) {
+          if (mfa.type !== MfaFactor.WebAuthn || mfa.id !== result.id) {
```
Comment (simeng-li) [inline packages/core/src/routes/interaction/verifications/mfa-payload-verification.ts:235]: Is this related to the `lastUseTime` updates? Don't have much context for this update.

File: packages/core/src/routes/interaction/verifications/mfa-verification.ts [hunk 3]
```diff
@@ -56,6 +61,39 @@ export const verifyMfa = async (
...
+      return verification.codes.some((code) => !code.usedAt);
+    })
+    .reduce<MfaVerification[]>((factors, verification) => {
+      // Ingnore duplicated verification
+      if (factors.some(({ type }) => type === verification.type)) {
+        return factors;
+      }
+
+      return [...factors, verification];
+    }, [])
```
Comment (simeng-li) [inline packages/core/src/routes/interaction/verifications/mfa-verification.ts:80]: could merge using one single reduce?

File: packages/core/src/routes/interaction/verifications/mfa-payload-verification.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/interaction/verifications/mfa-payload-verification.ts:235]: This is the WebAuthn's "counter" updates for some security device.

**Review by darcyYe — APPROVED** [review]

File: packages/core/src/routes/interaction/verifications/mfa-payload-verification.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/interaction/verifications/mfa-payload-verification.ts:235]: Year, it's a bug fix that should not be present in this PR, but since is tiny...

File: packages/core/src/routes/interaction/verifications/mfa-verification.ts (hunk 3 above)
Comment (wangsijie) [reply packages/core/src/routes/interaction/verifications/mfa-verification.ts]: I believe by spilting "filter" and "reduce", it would be easier to read the code.

File: packages/core/src/routes/interaction/mfa.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/interaction/mfa.ts:139]: created LOG-7236 to optimize.

**Review by simeng-li — APPROVED** [review]
LGTM
